code/modules/mob/mob.dm

```diff
@@ -1137,18 +1137,84 @@
 	if(client.mouse_override_icon)
 		client.mouse_pointer_icon = client.mouse_override_icon
 
-
-///This mob is abile to read books
+/// Is this mob able to read and write
 /mob/proc/is_literate()
 	return FALSE
-///Can this mob read (is literate and not blind)
-/mob/proc/can_read(obj/O)
+
+/// Is this mob affected by nearsight
+/mob/proc/is_nearsighted()
+	return FALSE
+
+/* Can this mob see in the dark
+ *
+ * This checks all traits, glasses, and robotic eyeball implants to see if the mob can see in the dark
+ * this does NOT check if the mob is missing it's eyeballs. Also see_in_dark is a BYOND mob var (that defaults to 2)
+**/
+/mob/proc/has_nightvision()
+	return see_in_dark >= NIGHTVISION_FOV_RANGE
+
+/* Checks if there is enough light where the mob is located
+ *
+ * Args:
+ *  light_amount (optional) - A decimal amount between 1.0 through 0.0 (default is 0.2)
+**/
+/mob/proc/has_light_nearby(light_amount = LIGHTING_TILE_IS_DARK)
+	var/turf/mob_location = get_turf(src)
+	return mob_location.get_lumcount() > light_amount
+
+/// Can this mob write
+/mob/proc/can_write(obj/writing_instrument)
+	if(!istype(writing_instrument, /obj/item/pen) && !istype(writing_instrument, /obj/item/toy/crayon))
+		to_chat(src, span_warning("You can't write with the [writing_instrument]!"))
+
 	if(is_blind())
-		to_chat(src, span_warning("As you are trying to read [O], you suddenly feel very stupid!"))
-		return
+		to_chat(src, span_warning("You are blind and can't write anything!"))
+		return FALSE
+
+	if(is_nearsighted())
+		to_chat(src, span_warning("Your vision is too blurry to write anything!"))
+		return FALSE
+
 	if(!is_literate())
-		to_chat(src, span_notice("You try to read [O], but can't comprehend any of it."))
-		return
+		to_chat(src, span_warning("You try to write, but don't know how to spell anything!"))
+		return FALSE
+
+	if(!has_light_nearby() && !has_nightvision())
+		to_chat(src, span_warning("It's too dark in here to write anything!"))
+		return FALSE
+
+	var/obj/item/pen/pen = writing_instrument
+	var/writing_instrument_requires_gravity
+
+	if(istype(pen))
+		writing_instrument_requires_gravity = pen.requires_gravity
+	else if(istype(writing_instrument, /obj/item/toy/crayon))
+		writing_instrument_requires_gravity = FALSE
+
+	if(!has_gravity() && writing_instrument_requires_gravity)
+		to_chat(src, span_warning("You try to write, but the [writing_instrument] doesn't work in zero gravity!"))
+		return FALSE
+
+	return TRUE
+
+/// Can this mob read
+/mob/proc/can_read(obj/O, check_for_light = TRUE)
+	if(is_blind())
+		to_chat(src, span_warning("You are blind and can't read anything!"))
+		return FALSE
+
+	if(is_nearsighted())
```

Comment:
this doesnt make any sense nearsighted just means far away things are blurry but close things arent. why would that stop you from reading a book